### ISSUE TITLE
Is watchexec supposed to do an initial run of the command no matter what? I tried limiting to create/modify, no difference

### ISSUE THREAD

**Issue by Jieiku** [watchexec/watchexec#649]
## watchexec --version

```
watchexec 1.23.0 (19a3fc9 2023-08-30) 
commit-hash: 19a3fc918948654ba0e2ced51fec2c8f4b78cd31
commit-date: 2023-08-30
build-date: 2023-08-30
release: 1.23.0
features:
```

## simple command that does it:

```bash

/usr/bin/watchexec -w /etc/letsencrypt/live --project-origin /etc/letsencrypt/live --fs-events create,modify -- date
```

### command that I was hoping to run:

```bash

/usr/bin/watchexec -w /etc/letsencrypt/live --project-origin /etc/letsencrypt/live --delay-run 9 --fs-events create,modify -o=do-nothing -f fullchain.pem -- date
```

*running the command `date` was just to test, later I plan to execute a shell script to reload postfix and dovecot services.*

## Notes:

The very second I run watchexec it prints the date, no delay:

![2023-08-31_14-56-49](https://github.com/watchexec/watchexec/assets/106644/c8a20aa5-d0a3-4162-a796-ce0b047690ef)

watchexec does still function, I get a new date everytime I push a new cert, its just the initial run of the date command that has me confused.

## log:

```json
{"timestamp":"2023-08-31T21:46:21.408777Z","level":"INFO","fields":{"message":"logging initialised"},"target":"watchexec_cli"}
{"timestamp":"2023-08-31T21:46:21.408874Z","level":"DEBUG","fields":{"message":"arguments","args":"Args { command: [\"date\"], paths: [\"/etc/letsencrypt/live\"], screen_clear: None, on_busy_update: Queue, watch_when_idle: false, restart: false, signal: None, kill: false, stop_signal: None, stop_timeout: TimeSpan(60s), debounce: TimeSpan(50ms), stdin_quit: false, no_vcs_ignore: false, no_project_ignore: false, no_global_ignore: false, no_default_ignore: false, no_discover_ignore: false, postpone: false, delay_run: None, poll: None, shell: None, no_shell: false, no_shell_long: false, no_environment: false, emit_events_to: Environment, env: [], no_process_group: false, once: false, notify: false, project_origin: Some(\"/etc/letsencrypt/live\"), workdir: None, filter_extensions: [], filter_patterns: [], filter_files: [], ignore_patterns: [], ignore_files: [], filter_fs_events: [Create, Modify], filter_fs_meta: false, print_events: false, verbose: Some(3), log_file: Some(\".\"), manual: false, completions: None }"},"target":"watchexec_cli"}
{"timestamp":"2023-08-31T21:46:21.409002Z","level":"INFO","fields":{"message":"constructing Watchexec from CLI","version":"1.23.0"},"target":"watchexec_cli"}
{"timestamp":"2023-08-31T21:46:21.409031Z","level":"DEBUG","fields":{"message":"InitConfig: on_error"},"target":"watchexec::config"}
{"timestamp":"2023-08-31T21:46:21.409395Z","level":"DEBUG","fields":{"message":"new"},"target":"watchexec_cli::config::runtime","span":{"name":"args-runtime"},"spans":[]}
{"timestamp":"2023-08-31T21:46:21.409463Z","level":"DEBUG","fields":{"message":"RuntimeConfig: command","command":"Shell { shell: Unix(\"sh\"), args: [], command: \"date\" }"},"target":"watchexec::config","span":{"name":"args-runtime"},"spans":[{"name":"args-runtime"}]}
{"timestamp":"2023-08-31T21:46:21.409522Z","level":"DEBUG","fields":{"message":"RuntimeConfig: pathset","pathset":"[WatchedPath(\"/etc/letsencrypt/live\")]"},"target":"watchexec::config","span":{"name":"args-runtime"},"spans":[{"name":"args-runtime"}]}
{"timestamp":"2023-08-31T21:46:21.409571Z","level":"DEBUG","fields":{"message":"RuntimeConfig: throttle","throttle":"50ms"},"target":"watchexec::config","span":{"name":"args-runtime"},"spans":[{"name":"args-runtime"}]}
{"timestamp":"2023-08-31T21:46:21.409603Z","level":"DEBUG","fields":{"message":"RuntimeConfig: command_grouped","grouped":"true"},"target":"watchexec::config","span":{"name":"args-runtime"},"spans":[{"name":"args-runtime"}]}
{"timestamp":"2023-08-31T21:46:21.409644Z","level":"DEBUG","fields":{"message":"RuntimeConfig: on_action"},"target":"watchexec::config","span":{"name":"args-runtime"},"spans":[{"name":"args-runtime"}]}
{"timestamp":"2023-08-31T21:46:21.409674Z","level":"DEBUG","fields":{"message":"additional environment variables to add to command","add_envs":"{}"},"target":"watchexec_cli::config::runtime","span":{"name":"args-runtime"},"spans":[{"name":"args-runtime"}]}
{"timestamp":"2023-08-31T21:46:21.409715Z","level":"DEBUG","fields":{"message":"RuntimeConfig: on_pre_spawn"},"target":"watchexec::config","span":{"name":"args-runtime"},"spans":[{"name":"args-runtime"}]}
{"timestamp":"2023-08-31T21:46:21.409758Z","level":"DEBUG","fields":{"message":"RuntimeConfig: on_post_spawn"},"target":"watchexec::config","span":{"name":"args-runtime"},"spans":[{"name":"args-runtime"}]}
{"timestamp":"2023-08-31T21:46:21.409793Z","level":"DEBUG","fields":{"message":"close","time.busy":"333µs","time.idle":"65.1µs"},"target":"watchexec_cli::config::runtime","span":{"name":"args-runtime"},"spans":[]}
{"timestamp":"2023-08-31T21:46:21.409991Z","level":"DEBUG","fields":{"message":"current directory","curdir":"\"/home/jake\""},"target":"watchexec_cli::filterer::common"}
{"timestamp":"2023-08-31T21:46:21.410033Z","level":"DEBUG","fields":{"message":"project origin override","origin":"\"/etc/letsencrypt/live\""},"target":"watchexec_cli::filterer::common"}
{"timestamp":"2023-08-31T21:46:21.410108Z","level":"INFO","fields":{"message":"resolved common/project origin","project_origin":"\"/etc/letsencrypt/live\""},"target":"watchexec_cli::filterer::common"}
{"timestamp":"2023-08-31T21:46:21.410160Z","level":"INFO","fields":{"message":"resolved working directory","workdir":"\"/home/jake\""},"target":"watchexec_cli::filterer::common"}
{"timestamp":"2023-08-31T21:46:21.410418Z","level":"INFO","fields":{"message":"resolved vcs types","vcs_types":"[]"},"target":"watchexec_cli::filterer::common"}
{"timestamp":"2023-08-31T21:46:21.411267Z","level":"DEBUG","fields":{"message":"discovered ignore files from project origin","ignores":"[]"},"target":"watchexec_cli::filterer::common"}
{"timestamp":"2023-08-31T21:46:21.411534Z","level":"DEBUG","fields":{"message":"discovered ignore files from environment","global_ignores":"[]"},"target":"watchexec_cli::filterer::common"}
{"timestamp":"2023-08-31T21:46:21.411576Z","level":"DEBUG","fields":{"message":"combined and applied overall vcs filter over ignores","ignores":"[]","vcs_types":"[]"},"target":"watchexec_cli::filterer::common"}
{"timestamp":"2023-08-31T21:46:21.411618Z","level":"DEBUG","fields":{"message":"combined with ignore files from command line / env","ignores":"[]","args.ignore_files":"[]"},"target":"watchexec_cli::filterer::common"}
{"timestamp":"2023-08-31T21:46:21.411646Z","level":"INFO","fields":{"message":"found some ignores","files":"[]"},"target":"watchexec_cli::filterer::common"}
{"timestamp":"2023-08-31T21:46:21.411679Z","level":"INFO","fields":{"message":"initialising Globset filterer"},"target":"watchexec_cli::filterer::globset"}
{"timestamp":"2023-08-31T21:46:21.411840Z","level":"DEBUG","fields":{"message":"glob converted to regex: Glob { glob: \"**/*.py[co]\", re: \"(?-u)^(?:/?|.*/)[^/]*\\\\.py[co]$\", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true, empty_alternates: false }, tokens: Tokens([RecursivePrefix, ZeroOrMore, Literal('.'), Literal('p'), Literal('y'), Class { negated: false, ranges: [('c', 'c'), ('o', 'o')] }]) }","log.target":"globset","log.module_path":"globset","log.file":"/home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/globset-0.4.13/src/lib.rs","log.line":426},"target":"globset"}
{"timestamp":"2023-08-31T21:46:21.411918Z","level":"DEBUG","fields":{"message":"glob converted to regex: Glob { glob: \"**/.#*\", re: \"(?-u)^(?:/?|.*/)\\\\.\\\\#[^/]*$\", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true, empty_alternates: false }, tokens: Tokens([RecursivePrefix, Literal('.'), Literal('#'), ZeroOrMore]) }","log.target":"globset","log.module_path":"globset","log.file":"/home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/globset-0.4.13/src/lib.rs","log.line":426},"target":"globset"}
{"timestamp":"2023-08-31T21:46:21.411980Z","level":"DEBUG","fields":{"message":"glob converted to regex: Glob { glob: \"**/.*.sw?\", re: \"(?-u)^(?:/?|.*/)\\\\.[^/]*\\\\.sw[^/]$\", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true, empty_alternates: false }, tokens: Tokens([RecursivePrefix, Literal('.'), ZeroOrMore, Literal('.'), Literal('s'), Literal('w'), Any]) }","log.target":"globset","log.module_path":"globset","log.file":"/home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/globset-0.4.13/src/lib.rs","log.line":426},"target":"globset"}
{"timestamp":"2023-08-31T21:46:21.412040Z","level":"DEBUG","fields":{"message":"glob converted to regex: Glob { glob: \"**/.*.sw?x\", re: \"(?-u)^(?:/?|.*/)\\\\.[^/]*\\\\.sw[^/]x$\", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true, empty_alternates: false }, tokens: Tokens([RecursivePrefix, Literal('.'), ZeroOrMore, Literal('.'), Literal('s'), Literal('w'), Any, Literal('x')]) }","log.target":"globset","log.module_path":"globset","log.file":"/home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/globset-0.4.13/src/lib.rs","log.line":426},"target":"globset"}
{"timestamp":"2023-08-31T21:46:21.412099Z","level":"DEBUG","fields":{"message":"glob converted to regex: Glob { glob: \"**/.bzr/**/*\", re: \"(?-u)^(?:/?|.*/)\\\\.bzr(?:/|/.*/)[^/]*$\", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true, empty_alternates: false }, tokens: Tokens([RecursivePrefix, Literal('.'), Literal('b'), Literal('z'), Literal('r'), RecursiveZeroOrMore, ZeroOrMore]) }","log.target":"globset","log.module_path":"globset","log.file":"/home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/globset-0.4.13/src/lib.rs","log.line":426},"target":"globset"}
{"timestamp":"2023-08-31T21:46:21.412160Z","level":"DEBUG","fields":{"message":"glob converted to regex: Glob { glob: \"**/_darcs/**/*\", re: \"(?-u)^(?:/?|.*/)_darcs(?:/|/.*/)[^/]*$\", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true, empty_alternates: false }, tokens: Tokens([RecursivePrefix, Literal('_'), Literal('d'), Literal('a'), Literal('r'), Literal('c'), Literal('s'), RecursiveZeroOrMore, ZeroOrMore]) }","log.target":"globset","log.module_path":"globset","log.file":"/home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/globset-0.4.13/src/lib.rs","log.line":426},"target":"globset"}
{"timestamp":"2023-08-31T21:46:21.412221Z","level":"DEBUG","fields":{"message":"glob converted to regex: Glob { glob: \"**/.fossil-settings/**/*\", re: \"(?-u)^(?:/?|.*/)\\\\.fossil\\\\-settings(?:/|/.*/)[^/]*$\", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true, empty_alternates: false }, tokens: Tokens([RecursivePrefix, Literal('.'), Literal('f'), Literal('o'), Literal('s'), Literal('s'), Literal('i'), Literal('l'), Literal('-'), Literal('s'), Literal('e'), Literal('t'), Literal('t'), Literal('i'), Literal('n'), Literal('g'), Literal('s'), RecursiveZeroOrMore, ZeroOrMore]) }","log.target":"globset","log.module_path":"globset","log.file":"/home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/globset-0.4.13/src/lib.rs","log.line":426},"target":"globset"}
{"timestamp":"2023-08-31T21:46:21.412289Z","level":"DEBUG","fields":{"message":"glob converted to regex: Glob { glob: \"**/.git/**/*\", re: \"(?-u)^(?:/?|.*/)\\\\.git(?:/|/.*/)[^/]*$\", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true, empty_alternates: false }, tokens: Tokens([RecursivePrefix, Literal('.'), Literal('g'), Literal('i'), Literal('t'), RecursiveZeroOrMore, ZeroOrMore]) }","log.target":"globset","log.module_path":"globset","log.file":"/home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/globset-0.4.13/src/lib.rs","log.line":426},"target":"globset"}
{"timestamp":"2023-08-31T21:46:21.412349Z","level":"DEBUG","fields":{"message":"glob converted to regex: Glob { glob: \"**/.hg/**/*\", re: \"(?-u)^(?:/?|.*/)\\\\.hg(?:/|/.*/)[^/]*$\", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true, empty_alternates: false }, tokens: Tokens([RecursivePrefix, Literal('.'), Literal('h'), Literal('g'), RecursiveZeroOrMore, ZeroOrMore]) }","log.target":"globset","log.module_path":"globset","log.file":"/home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/globset-0.4.13/src/lib.rs","log.line":426},"target":"globset"}
{"timestamp":"2023-08-31T21:46:21.412409Z","level":"DEBUG","fields":{"message":"glob converted to regex: Glob { glob: \"**/.pijul/**/*\", re: \"(?-u)^(?:/?|.*/)\\\\.pijul(?:/|/.*/)[^/]*$\", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true, empty_alternates: false }, tokens: Tokens([RecursivePrefix, Literal('.'), Literal('p'), Literal('i'), Literal('j'), Literal('u'), Literal('l'), RecursiveZeroOrMore, ZeroOrMore]) }","log.target":"globset","log.module_path":"globset","log.file":"/home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/globset-0.4.13/src/lib.rs","log.line":426},"target":"globset"}
{"timestamp":"2023-08-31T21:46:21.412466Z","level":"DEBUG","fields":{"message":"glob converted to regex: Glob { glob: \"**/.svn/**/*\", re: \"(?-u)^(?:/?|.*/)\\\\.svn(?:/|/.*/)[^/]*$\", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true, empty_alternates: false }, tokens: Tokens([RecursivePrefix, Literal('.'), Literal('s'), Literal('v'), Literal('n'), RecursiveZeroOrMore, ZeroOrMore]) }","log.target":"globset","log.module_path":"globset","log.file":"/home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/globset-0.4.13/src/lib.rs","log.line":426},"target":"globset"}
{"timestamp":"2023-08-31T21:46:21.412528Z","level":"DEBUG","fields":{"message":"built glob set; 0 literals, 1 basenames, 0 extensions, 0 prefixes, 0 suffixes, 2 required extensions, 11 regexes","log.target":"globset","log.module_path":"globset","log.file":"/home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/globset-0.4.13/src/lib.rs","log.line":431},"target":"globset"}
{"timestamp":"2023-08-31T21:46:21.414021Z","level":"DEBUG","fields":{"message":"globset filterer built","origin":"\"/etc/letsencrypt/live\"","num_filters":"0","num_neg_filters":"0","num_ignores":"14","num_in_ignore_files":"(0, 0)","num_neg_ignores":"0","num_extensions":"0"},"target":"watchexec_filterer_globset"}
{"timestamp":"2023-08-31T21:46:21.414129Z","level":"DEBUG","fields":{"message":"RuntimeConfig: filterer","filterer":"WatchexecFilterer { inner: GlobsetFilterer { origin: \"/etc/letsencrypt/live\", filters: Gitignore { set: GlobSet { len: 0, strats: [] }, root: \"/etc/letsencrypt/live\", globs: [], num_ignores: 0, num_whitelists: 0, matches: Some(ThreadLocal { local_data: None }) }, ignores: Gitignore { set: GlobSet { len: 14, strats: [Extension(ExtensionStrategy({})), BasenameLiteral(BasenameLiteralStrategy({[46, 68, 83, 95, 83, 116, 111, 114, 101]: [0]})), Literal(LiteralStrategy({})), Suffix(SuffixStrategy { matcher: AhoCorasick(dfa::DFA(\nD 000000: \\x00 => 0\nF 000001:\n >000002: \\x00 => 2\n  000003: \\x00 => 0\nmatch kind: Standard\nprefilter: false\nstate length: 4\npattern length: 0\nshortest pattern length: 18446744073709551615\nlongest pattern length: 0\nalphabet length: 1\nstride: 1\nbyte classes: ByteClasses(0 => [0-255])\nmemory usage: 16\n)\n), map: [], longest: 0 }), Prefix(PrefixStrategy { matcher: AhoCorasick(dfa::DFA(\nD 000000: \\x00 => 0\nF 000001:\n >000002: \\x00 => 2\n  000003: \\x00 => 0\nmatch kind: Standard\nprefilter: false\nstate length: 4\npattern length: 0\nshortest pattern length: 18446744073709551615\nlongest pattern length: 0\nalphabet length: 1\nstride: 1\nbyte classes: ByteClasses(0 => [0-255])\nmemory usage: 16\n)\n), map: [], longest: 0 }), RequiredExtension(RequiredExtensionStrategy({[46, 108, 111, 103]: [(1, Regex(\"(?-u)^(?:/?|.*/)watchexec\\\\.[^/]*\\\\.log$\"))], [46, 107, 97, 116, 101, 45, 115, 119, 112]: [(4, Regex(\"(?-u)^(?:/?|.*/)\\\\.[^/]*\\\\.kate\\\\-swp$\"))]})), Regex(RegexSetStrategy { matcher: RegexSet([\"(?-u)^(?:/?|.*/)[^/]*\\\\.py[co]$\", \"(?-u)^(?:/?|.*/)\\\\.\\\\#[^/]*$\", \"(?-u)^(?:/?|.*/)\\\\.[^/]*\\\\.sw[^/]$\", \"(?-u)^(?:/?|.*/)\\\\.[^/]*\\\\.sw[^/]x$\", \"(?-u)^(?:/?|.*/)\\\\.bzr(?:/|/.*/)[^/]*$\", \"(?-u)^(?:/?|.*/)_darcs(?:/|/.*/)[^/]*$\", \"(?-u)^(?:/?|.*/)\\\\.fossil\\\\-settings(?:/|/.*/)[^/]*$\", \"(?-u)^(?:/?|.*/)\\\\.git(?:/|/.*/)[^/]*$\", \"(?-u)^(?:/?|.*/)\\\\.hg(?:/|/.*/)[^/]*$\", \"(?-u)^(?:/?|.*/)\\\\.pijul(?:/|/.*/)[^/]*$\", \"(?-u)^(?:/?|.*/)\\\\.svn(?:/|/.*/)[^/]*$\"]), map: [2, 3, 5, 6, 7, 8, 9, 10, 11, 12, 13] })] }, root: \"/etc/letsencrypt/live\", globs: [Glob { from: None, original: \"**/.DS_Store\", actual: \"**/.DS_Store\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \"watchexec.*.log\", actual: \"**/watchexec.*.log\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \"*.py[co]\", actual: \"**/*.py[co]\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \".#*\", actual: \"**/.#*\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \".*.kate-swp\", actual: \"**/.*.kate-swp\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \".*.sw?\", actual: \"**/.*.sw?\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \".*.sw?x\", actual: \"**/.*.sw?x\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \"**/.bzr/**\", actual: \"**/.bzr/**/*\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \"**/_darcs/**\", actual: \"**/_darcs/**/*\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \"**/.fossil-settings/**\", actual: \"**/.fossil-settings/**/*\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \"**/.git/**\", actual: \"**/.git/**/*\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \"**/.hg/**\", actual: \"**/.hg/**/*\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \"**/.pijul/**\", actual: \"**/.pijul/**/*\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \"**/.svn/**\", actual: \"**/.svn/**/*\", is_whitelist: false, is_only_dir: false }], num_ignores: 14, num_whitelists: 0, matches: Some(ThreadLocal { local_data: None }) }, ignore_files: IgnoreFilterer(IgnoreFilter { origin: \"/etc/letsencrypt/live\", ignores: Trie { length: 1, node: TrieNode { key: NibbleVec [], key_value: None, child_count: 1, children: [None, None, Some(TrieNode { key: NibbleVec [2, 15], key_value: Some(KeyValue { key: \"/\", value: Ignore { gitignore: Gitignore { set: GlobSet { len: 0, strats: [] }, root: \"\", globs: [], num_ignores: 0, num_whitelists: 0, matches: None }, builder: None } }), child_count: 0, children: [None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None] }), None, None, None, None, None, None, None, None, None, None, None, None, None] } } }), extensions: [] }, fs_events: [Create, Modify] }"},"target":"watchexec::config"}
{"timestamp":"2023-08-31T21:46:21.414460Z","level":"INFO","fields":{"message":"initialising Watchexec runtime"},"target":"watchexec_cli"}
{"timestamp":"2023-08-31T21:46:21.414490Z","level":"DEBUG","fields":{"message":"initialising","init":"InitConfig { error_channel_size: 64, event_channel_size: 1024, .. }","runtime":"RuntimeConfig { fs: WorkingData { pathset: [WatchedPath(\"/etc/letsencrypt/live\")], watcher: Native }, keyboard: WorkingData { eof: false }, action: WorkingData { throttle: 50ms, commands: [Shell { shell: Unix(\"sh\"), args: [], command: \"date\" }], grouped: true, filterer: WatchexecFilterer { inner: GlobsetFilterer { origin: \"/etc/letsencrypt/live\", filters: Gitignore { set: GlobSet { len: 0, strats: [] }, root: \"/etc/letsencrypt/live\", globs: [], num_ignores: 0, num_whitelists: 0, matches: Some(ThreadLocal { local_data: None }) }, ignores: Gitignore { set: GlobSet { len: 14, strats: [Extension(ExtensionStrategy({})), BasenameLiteral(BasenameLiteralStrategy({[46, 68, 83, 95, 83, 116, 111, 114, 101]: [0]})), Literal(LiteralStrategy({})), Suffix(SuffixStrategy { matcher: AhoCorasick(dfa::DFA(\nD 000000: \\x00 => 0\nF 000001:\n >000002: \\x00 => 2\n  000003: \\x00 => 0\nmatch kind: Standard\nprefilter: false\nstate length: 4\npattern length: 0\nshortest pattern length: 18446744073709551615\nlongest pattern length: 0\nalphabet length: 1\nstride: 1\nbyte classes: ByteClasses(0 => [0-255])\nmemory usage: 16\n)\n), map: [], longest: 0 }), Prefix(PrefixStrategy { matcher: AhoCorasick(dfa::DFA(\nD 000000: \\x00 => 0\nF 000001:\n >000002: \\x00 => 2\n  000003: \\x00 => 0\nmatch kind: Standard\nprefilter: false\nstate length: 4\npattern length: 0\nshortest pattern length: 18446744073709551615\nlongest pattern length: 0\nalphabet length: 1\nstride: 1\nbyte classes: ByteClasses(0 => [0-255])\nmemory usage: 16\n)\n), map: [], longest: 0 }), RequiredExtension(RequiredExtensionStrategy({[46, 108, 111, 103]: [(1, Regex(\"(?-u)^(?:/?|.*/)watchexec\\\\.[^/]*\\\\.log$\"))], [46, 107, 97, 116, 101, 45, 115, 119, 112]: [(4, Regex(\"(?-u)^(?:/?|.*/)\\\\.[^/]*\\\\.kate\\\\-swp$\"))]})), Regex(RegexSetStrategy { matcher: RegexSet([\"(?-u)^(?:/?|.*/)[^/]*\\\\.py[co]$\", \"(?-u)^(?:/?|.*/)\\\\.\\\\#[^/]*$\", \"(?-u)^(?:/?|.*/)\\\\.[^/]*\\\\.sw[^/]$\", \"(?-u)^(?:/?|.*/)\\\\.[^/]*\\\\.sw[^/]x$\", \"(?-u)^(?:/?|.*/)\\\\.bzr(?:/|/.*/)[^/]*$\", \"(?-u)^(?:/?|.*/)_darcs(?:/|/.*/)[^/]*$\", \"(?-u)^(?:/?|.*/)\\\\.fossil\\\\-settings(?:/|/.*/)[^/]*$\", \"(?-u)^(?:/?|.*/)\\\\.git(?:/|/.*/)[^/]*$\", \"(?-u)^(?:/?|.*/)\\\\.hg(?:/|/.*/)[^/]*$\", \"(?-u)^(?:/?|.*/)\\\\.pijul(?:/|/.*/)[^/]*$\", \"(?-u)^(?:/?|.*/)\\\\.svn(?:/|/.*/)[^/]*$\"]), map: [2, 3, 5, 6, 7, 8, 9, 10, 11, 12, 13] })] }, root: \"/etc/letsencrypt/live\", globs: [Glob { from: None, original: \"**/.DS_Store\", actual: \"**/.DS_Store\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \"watchexec.*.log\", actual: \"**/watchexec.*.log\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \"*.py[co]\", actual: \"**/*.py[co]\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \".#*\", actual: \"**/.#*\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \".*.kate-swp\", actual: \"**/.*.kate-swp\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \".*.sw?\", actual: \"**/.*.sw?\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \".*.sw?x\", actual: \"**/.*.sw?x\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \"**/.bzr/**\", actual: \"**/.bzr/**/*\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \"**/_darcs/**\", actual: \"**/_darcs/**/*\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \"**/.fossil-settings/**\", actual: \"**/.fossil-settings/**/*\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \"**/.git/**\", actual: \"**/.git/**/*\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \"**/.hg/**\", actual: \"**/.hg/**/*\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \"**/.pijul/**\", actual: \"**/.pijul/**/*\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \"**/.svn/**\", actual: \"**/.svn/**/*\", is_whitelist: false, is_only_dir: false }], num_ignores: 14, num_whitelists: 0, matches: Some(ThreadLocal { local_data: None }) }, ignore_files: IgnoreFilterer(IgnoreFilter { origin: \"/etc/letsencrypt/live\", ignores: Trie { length: 1, node: TrieNode { key: NibbleVec [], key_value: None, child_count: 1, children: [None, None, Some(TrieNode { key: NibbleVec [2, 15], key_value: Some(KeyValue { key: \"/\", value: Ignore { gitignore: Gitignore { set: GlobSet { len: 0, strats: [] }, root: \"\", globs: [], num_ignores: 0, num_whitelists: 0, matches: None }, builder: None } }), child_count: 0, children: [None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None] }), None, None, None, None, None, None, None, None, None, None, None, None, None] } } }), extensions: [] }, fs_events: [Create, Modify] }, .. } }","pid":"1356","version":"2.3.0"},"target":"watchexec::watchexec"}
{"timestamp":"2023-08-31T21:46:21.414833Z","level":"DEBUG","fields":{"message":"kicking off with empty event"},"target":"watchexec_cli"}
{"timestamp":"2023-08-31T21:46:21.414895Z","level":"INFO","fields":{"message":"running main loop"},"target":"watchexec_cli"}
{"timestamp":"2023-08-31T21:46:21.414929Z","level":"DEBUG","fields":{"message":"handing over main task handle"},"target":"watchexec::watchexec"}
{"timestamp":"2023-08-31T21:46:21.414967Z","level":"DEBUG","fields":{"message":"starting main task"},"target":"watchexec::watchexec"}
{"timestamp":"2023-08-31T21:46:21.415060Z","level":"DEBUG","fields":{"message":"spawning subtask","subtask":"action"},"target":"watchexec::watchexec"}
{"timestamp":"2023-08-31T21:46:21.415132Z","level":"DEBUG","fields":{"message":"spawning subtask","subtask":"fs"},"target":"watchexec::watchexec"}
{"timestamp":"2023-08-31T21:46:21.415175Z","level":"DEBUG","fields":{"message":"spawning subtask","subtask":"signal"},"target":"watchexec::watchexec"}
{"timestamp":"2023-08-31T21:46:21.415217Z","level":"DEBUG","fields":{"message":"spawning subtask","subtask":"keyboard"},"target":"watchexec::watchexec"}
{"timestamp":"2023-08-31T21:46:21.415251Z","level":"DEBUG","fields":{"message":"spawning subtask","subtask":"error_hook"},"target":"watchexec::watchexec"}
{"timestamp":"2023-08-31T21:46:21.415259Z","level":"INFO","fields":{"message":"action constructed","action":"Action { events: [Event { tags: [], metadata: {} }], outcome: OnceCell(Uninit) }"},"target":"watchexec::action::worker"}
{"timestamp":"2023-08-31T21:46:21.415348Z","level":"DEBUG","fields":{"message":"launching unix signal worker"},"target":"watchexec::signal"}
{"timestamp":"2023-08-31T21:46:21.415374Z","level":"DEBUG","fields":{"message":"running action handler"},"target":"watchexec::action::worker"}
{"timestamp":"2023-08-31T21:46:21.415351Z","level":"DEBUG","fields":{"message":"launching filesystem worker"},"target":"watchexec::fs"}
{"timestamp":"2023-08-31T21:46:21.415502Z","level":"DEBUG","fields":{"message":"action handler finished","outcome":"IfRunning(Both(Wait, Start), Start)"},"target":"watchexec::action::worker"}
{"timestamp":"2023-08-31T21:46:21.415580Z","level":"DEBUG","fields":{"message":"creating new watcher","kind":"Native"},"target":"watchexec::fs"}
{"timestamp":"2023-08-31T21:46:21.415620Z","level":"INFO","fields":{"message":"outcome resolved","outcome":"Start"},"target":"watchexec::action::worker"}
{"timestamp":"2023-08-31T21:46:21.415739Z","level":"DEBUG","fields":{"message":"spawning outcome worker","outcome":"Start","gen":"1"},"target":"watchexec::action::outcome_worker"}
{"timestamp":"2023-08-31T21:46:21.415772Z","level":"DEBUG","fields":{"message":"action process done"},"target":"watchexec::action::worker"}
{"timestamp":"2023-08-31T21:46:21.415866Z","level":"DEBUG","fields":{"message":"applying changes to the watcher","to_watch":"[WatchedPath(\"/etc/letsencrypt/live\")]","to_drop":"[]"},"target":"watchexec::fs"}
{"timestamp":"2023-08-31T21:46:21.415878Z","level":"DEBUG","fields":{"message":"outcome worker finished","outcome":"Start","gen":"1"},"target":"watchexec::action::outcome_worker"}
{"timestamp":"2023-08-31T21:46:21.416008Z","level":"DEBUG","fields":{"message":"new"},"target":"watchexec::command::supervisor","span":{"name":"supervisor"},"spans":[]}
{"timestamp":"2023-08-31T21:46:21.416071Z","level":"DEBUG","fields":{"message":"preparing command","grouped":"true","command":"Shell { shell: Unix(\"sh\"), args: [], command: \"date\" }"},"target":"watchexec::command::supervisor","span":{"name":"supervisor"},"spans":[{"name":"supervisor"}]}
{"timestamp":"2023-08-31T21:46:21.416120Z","level":"DEBUG","fields":{"message":"running pre-spawn handler"},"target":"watchexec::command::supervisor","span":{"name":"supervisor"},"spans":[{"name":"supervisor"}]}
{"timestamp":"2023-08-31T21:46:21.416187Z","level":"INFO","fields":{"message":"spawning command","command":"Command { std: \"sh\" \"-c\" \"date\", kill_on_drop: false }"},"target":"watchexec::command::supervisor","span":{"name":"supervisor"},"spans":[{"name":"supervisor"}]}
{"timestamp":"2023-08-31T21:46:21.416948Z","level":"DEBUG","fields":{"message":"resetting process sigmask","unblocking":"SigSet { sigset: sigset_t { __val: [18446744067267100670, 4096, 140503278424208, 275, 562949953421311, 94010312711326, 140503503324560, 0, 140503503324552, 562949953421311, 94010327911160, 18444492273895866368, 0, 94010312213400, 140503503324552, 72479572293063680] } }"},"target":"watchexec::command::supervisor","span":{"name":"supervisor"},"spans":[{"name":"supervisor"}]}
{"timestamp":"2023-08-31T21:46:21.417207Z","level":"DEBUG","fields":{"message":"sigmask reset","oldset":"SigSet { sigset: sigset_t { __val: [0, 1, 140503278435728, 94010327910600, 140503278432368, 140503278426320, 274877906944, 0, 140503503324560, 16655975262208267178, 140503509049408, 1, 0, 140501533589504, 140503509053640, 140503509053640] } }"},"target":"watchexec::command::supervisor","span":{"name":"supervisor"},"spans":[{"name":"supervisor"}]}
{"timestamp":"2023-08-31T21:46:21.418276Z","level":"INFO","fields":{"message":"process group spawned","pgid":"1363"},"target":"watchexec::command::supervisor","span":{"name":"supervisor"},"spans":[{"name":"supervisor"}]}
{"timestamp":"2023-08-31T21:46:21.418371Z","level":"DEBUG","fields":{"message":"running post-spawn handler"},"target":"watchexec::command::supervisor","span":{"name":"supervisor"},"spans":[{"name":"supervisor"}]}
{"timestamp":"2023-08-31T21:46:21.418427Z","level":"DEBUG","fields":{"message":"spawned process","process":"Grouped(AsyncGroupChild)","pid":"1363"},"target":"watchexec::command::supervisor","span":{"name":"supervisor"},"spans":[{"name":"supervisor"}]}
{"timestamp":"2023-08-31T21:46:21.421165Z","level":"DEBUG","fields":{"message":"creating synthetic process completion event","event":"Event { tags: [Source(Internal), ProcessCompletion(Some(Success))], metadata: {} }"},"target":"watchexec::command::supervisor","span":{"name":"supervisor"},"spans":[{"name":"supervisor"}]}
{"timestamp":"2023-08-31T21:46:21.421305Z","level":"DEBUG","fields":{"message":"no more commands to supervise"},"target":"watchexec::command::supervisor","span":{"name":"supervisor"},"spans":[{"name":"supervisor"}]}
{"timestamp":"2023-08-31T21:46:21.421376Z","level":"DEBUG","fields":{"message":"close","time.busy":"2.55ms","time.idle":"2.81ms"},"target":"watchexec::command::supervisor","span":{"name":"supervisor"},"spans":[]}
{"timestamp":"2023-08-31T21:46:21.472748Z","level":"INFO","fields":{"message":"action constructed","action":"Action { events: [Event { tags: [Source(Internal), ProcessCompletion(Some(Success))], metadata: {} }], outcome: OnceCell(Uninit) }"},"target":"watchexec::action::worker"}
{"timestamp":"2023-08-31T21:46:21.472821Z","level":"DEBUG","fields":{"message":"running action handler"},"target":"watchexec::action::worker"}
{"timestamp":"2023-08-31T21:46:21.472905Z","level":"DEBUG","fields":{"message":"action handler finished","outcome":"DoNothing"},"target":"watchexec::action::worker"}
{"timestamp":"2023-08-31T21:46:21.472937Z","level":"INFO","fields":{"message":"outcome resolved","outcome":"DoNothing"},"target":"watchexec::action::worker"}
{"timestamp":"2023-08-31T21:46:21.472961Z","level":"DEBUG","fields":{"message":"spawning outcome worker","outcome":"DoNothing","gen":"2"},"target":"watchexec::action::outcome_worker"}
{"timestamp":"2023-08-31T21:46:21.473005Z","level":"DEBUG","fields":{"message":"action process done"},"target":"watchexec::action::worker"}
{"timestamp":"2023-08-31T21:46:21.473043Z","level":"DEBUG","fields":{"message":"outcome worker finished","outcome":"DoNothing","gen":"2"},"target":"watchexec::action::outcome_worker"}
{"timestamp":"2023-08-31T21:46:22.805088Z","level":"DEBUG","fields":{"message":"received unix signal","sig":"Interrupt"},"target":"watchexec::signal"}
{"timestamp":"2023-08-31T21:46:22.805247Z","level":"INFO","fields":{"message":"action constructed","action":"Action { events: [Event { tags: [Source(Keyboard), Signal(Interrupt)], metadata: {} }], outcome: OnceCell(Uninit) }"},"target":"watchexec::action::worker"}
{"timestamp":"2023-08-31T21:46:22.805290Z","level":"DEBUG","fields":{"message":"running action handler"},"target":"watchexec::action::worker"}
{"timestamp":"2023-08-31T21:46:22.805354Z","level":"DEBUG","fields":{"message":"action handler finished","outcome":"Both(Stop, Exit)"},"target":"watchexec::action::worker"}
{"timestamp":"2023-08-31T21:46:22.805408Z","level":"INFO","fields":{"message":"outcome resolved","outcome":"Both(Stop, Exit)"},"target":"watchexec::action::worker"}
{"timestamp":"2023-08-31T21:46:22.805437Z","level":"DEBUG","fields":{"message":"spawning outcome worker","outcome":"Both(Stop, Exit)","gen":"3"},"target":"watchexec::action::outcome_worker"}
{"timestamp":"2023-08-31T21:46:22.805465Z","level":"DEBUG","fields":{"message":"action process done"},"target":"watchexec::action::worker"}
{"timestamp":"2023-08-31T21:46:22.805532Z","level":"INFO","fields":{"message":"propagating graceful exit","gen":"3"},"target":"watchexec::action::outcome_worker"}
{"timestamp":"2023-08-31T21:46:22.805596Z","level":"DEBUG","fields":{"message":"finishing subtask","subtask":"error_hook"},"target":"watchexec::watchexec"}
{"timestamp":"2023-08-31T21:46:22.805632Z","level":"DEBUG","fields":{"message":"aborting subtask","subtask":"error_hook"},"target":"watchexec::watchexec"}
{"timestamp":"2023-08-31T21:46:22.805685Z","level":"DEBUG","fields":{"message":"aborting subtask","subtask":"action"},"target":"watchexec::watchexec"}
{"timestamp":"2023-08-31T21:46:22.805710Z","level":"DEBUG","fields":{"message":"aborting subtask","subtask":"fs"},"target":"watchexec::watchexec"}
{"timestamp":"2023-08-31T21:46:22.805743Z","level":"DEBUG","fields":{"message":"aborting subtask","subtask":"signal"},"target":"watchexec::watchexec"}
{"timestamp":"2023-08-31T21:46:22.805790Z","level":"DEBUG","fields":{"message":"aborting subtask","subtask":"keyboard"},"target":"watchexec::watchexec"}
{"timestamp":"2023-08-31T21:46:22.805819Z","level":"DEBUG","fields":{"message":"main task graceful exit"},"target":"watchexec::watchexec"}
{"timestamp":"2023-08-31T21:46:22.805938Z","level":"INFO","fields":{"message":"done with main loop"},"target":"watchexec_cli"}
```

**Comment by passcod** [comment]
There's a `--postpone` flag for this:

> **-p, --postpone**
> 
> Wait until first change before running command
> 
> By default, Watchexec will run the command once immediately. With this option, it will instead wait until an
> event is detected before running the command as normal.

**Comment by Jieiku** [comment]
I sweat i read over the watchexec --help 3 times, but it just wasnt clicking that it was the fix I was looking for.

Thanks for a great program and the quick response!
